// stream-chat-react-native-core/lib/commonjs/components/Message/MessageSimple/MessageError.js
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MessageError = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireDefault(require("react"));
var _reactNative = require("react-native");
var _ThemeContext = require("../../../contexts/themeContext/ThemeContext");
var _icons = require("../../../icons");
var _this = this,
  _jsxFileName = "/home/runner/work/stream-chat-react-native/stream-chat-react-native/package/src/components/Message/MessageSimple/MessageError.tsx";
var MessageError = function MessageError(_ref) {
  var style = _ref.style;
  var _useTheme = (0, _ThemeContext.useTheme)(),
    _useTheme$theme = _useTheme.theme,
    accent_red = _useTheme$theme.colors.accent_red,
    _useTheme$theme$messa = _useTheme$theme.messageSimple.content,
    errorIcon = _useTheme$theme$messa.errorIcon,
    errorIconContainer = _useTheme$theme$messa.errorIconContainer;
  return _react["default"].createElement(_reactNative.View, {
    style: [_reactNative.StyleSheet.absoluteFill, style],
    testID: "message-error",
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 22,
      columnNumber: 5
    }
  }, _react["default"].createElement(_reactNative.View, {
    style: errorIconContainer,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 23,
      columnNumber: 7
    }
  }, _react["default"].createElement(_icons.Error, (0, _extends2["default"])({
    pathFill: accent_red
  }, errorIcon, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 24,
      columnNumber: 9
    }
  }))));
};
exports.MessageError = MessageError;
